src/slam_solver.h

```diff
@@ -0,0 +1,22 @@
+/* Extends ScanSolver to allow different solvers to publish graph visualizations differently */
+#ifndef SLAM_SOLVER_H
+#define SLAM_SOLVER_H
+
+#include <open_karto/Mapper.h>
+#include <visualization_msgs/MarkerArray.h>
+#include <string>
+
+namespace karto {
+
+class SLAMSolver : public karto::ScanSolver
```

Comment:
A better design would be to leave karto::ScanSolver as it is (don't extend it) and add a new base class karto::ScanSolverVisualizer that you can attach. You can then have different visualizers for the same solver.
The instantiation would look something like this:
```
solver_ = new SpaSolver(); // or plugin_loader_.createInstance(solver_name);
visualizer_ =  new SpaSolverVisualizer(); // or plugin_loader_.createInstance(solver_visualizer_name);
visualizer_->initialize(solver_);
//...
visualization_msgs::MarkerArray marray = visualizer_->createVisualizationMarkers();
```